include/nuttx/arch.h

```diff
@@ -99,6 +99,18 @@
 #define DEBUGPOINT_BREAKPOINT    0x04
 #define DEBUGPOINT_STEPPOINT     0x05
 
+/* Ensure that LOOPSPERMSEC is not the default, invalid value. If it is, let
+ * the user know that they need to change it and then calibrate it.
+ */
+
+static_assert(
+    CONFIG_BOARD_LOOPSPERMSEC >= 0,
```

Comment:
it's better to move the assert the place with really use it, since not all busy wait implementation depend on it.

Reply:
I just put it here since it will catch it where it's defined. Maybe I can put a guard around it so that it's not triggered if `ARCH_ARLAM` or `ARCH_TIMER` are enabled?

Reply:
> there are many method to implement the busy wait loop, it's better to put your assert to the place which really use this macro.

But this is used in many different places, so it would require replicating the assert statement many times

Reply:
this mean is time to merge the implementation into one:), since the soft busy wait isn't couple with arch/chip at all.

Reply:
> this mean is time to merge the implementation into one:), since the soft busy wait isn't couple with arch/chip at all.

This is done in #17071